File: src/main/java/com/mouad/IT_Support/model/Ticket.java
package com.mouad.IT_Support.model;


import com.mouad.IT_Support.model.enums.EtatTicket;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;


@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Ticket {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long ticketId;
    private String description;
    private LocalDate dateDeCreation;
    @Enumerated(EnumType.STRING)
    private EtatTicket statut;

    @ManyToOne
    @JoinColumn(name = "technicien_id")
    private Technicien technicien;

    @ManyToOne
    @JoinColumn(name = "utilisateur_id")
    private Utilisateur utilisateur;

    @ManyToOne
    @JoinColumn(name = "equipementId")
    private Equipment equipment;

    @ManyToOne
    @JoinColumn(name = "panneId")
    private Panne panne;

}

File: IT_Support-Angular/src/app/classes/ticket/ticket.ts
import {EtatTicket} from "../enums/etat-ticket";
import {Equipement} from "../equipement/equipement";
import {Panne} from "../panne/panne";
import {Technicien} from "../technicien/technicien";
import {Utilisateur} from "../utilisateur/utilisateur";

export class Ticket {
  ticketId!: number;
  technicien!: Technicien;
  utilisateur!: Utilisateur;
  description!: string;
  dateDeCreation!: string;
  EtatTicket!: EtatTicket;
  equipment!: Equipement;
  panne!: Panne;
}
